pkg/metrics/metricbuilder/metricbuilder.go

```diff
@@ -0,0 +1,64 @@
+package metricbuilder
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// MetricBuilder is the basic metric type that must be used by any potential metric added to
+// the system. Any metric added must have the following functions which would be used in the
+// workflow.
+type MetricBuilder struct {
+	labels         []string
+	labelKeyValues map[string]string
+	desc           string
+	name           string
+	value          float64
+	buckets        []float64
+	metricType     string
+}
+
+// build function creates the required prometheus collector object with the values
+// it hs in the MetricBuilder calling object.
+func (m MetricBuilder) build() prometheus.Collector {
+	switch m.metricType {
+	case "modification":
```

Comment:
similar to comment about `MetricBuilder.metricType` this should switch on the prometheus metric type, so there should be at most 3 cases or however many prometheus metrics types there are.